tracking/protocol.cpp

```diff
@@ -23,17 +48,54 @@ vector<uint8_t> Protocol::CreateAuthPacket(string const & clientId)
   return packet;
 }
 
-vector<uint8_t> Protocol::CreateDataPacket(DataElements const & points)
+vector<uint8_t> Protocol::CreateDataPacket(DataElementsCirc const & points)
 {
-  vector<uint8_t> buffer;
-  MemWriter<decltype(buffer)> writer(buffer);
-  Encoder::SerializeDataPoints(Encoder::kLatestVersion, writer, points);
+  return CreateDataPacketImpl(points);
+}
 
-  vector<uint8_t> packet;
-  InitHeader(packet, PacketType::CurrentData, static_cast<uint32_t>(buffer.size()));
-  packet.insert(packet.end(), begin(buffer), end(buffer));
+vector<uint8_t> Protocol::CreateDataPacket(DataElementsVec const & points)
+{
+  return CreateDataPacketImpl(points);
+}
 
-  return packet;
+pair<Protocol::PacketType, size_t> Protocol::DecodeHeader(vector<uint8_t> const & data)
+{
+  ASSERT_GREATER_OR_EQUAL(data.size(), sizeof(uint32_t /* header */), ());
+
+  uint32_t size = (*reinterpret_cast<uint32_t const *>(data.data())) & 0xFFFFFF00;
+  if (!IsBigEndian())
+    size = ReverseByteOrder(size);
+
+  return make_pair(PacketType(static_cast<uint8_t>(data[0])), size);
+}
+
+string Protocol::DecodeAuthPacket(Protocol::PacketType type, vector<uint8_t> const & data)
+{
+  ASSERT_GREATER_OR_EQUAL(data.size(), sizeof(uint32_t /* header */), ());
+  switch (type)
+  {
+  case Protocol::PacketType::AuthV0:
+    return string(begin(data) + sizeof(uint32_t /* header */), end(data));
+  case Protocol::PacketType::DataV0: break;
```

Comment:
Just return string() here instead of break. And get rid of return statement at the end.